### PR TITLE
clientv3: document possible memory pile-up in Watch

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/9103.

### CODE REVIEW

File: clientv3/watch.go [hunk 1]
```diff
@@ -50,11 +50,13 @@ type Watcher interface {
 	// and "WatchResponse" from this closed channel has zero events and nil "Err()".
 	// If the context is "context.Background/TODO", returned "WatchChan" will not be closed
 	// and wait until events happen, except when server returns a non-recoverable error.
-	// For example, when context passed with "WithRequireLeader" and the connected server
-	// has no leader, error "etcdserver: no leader" is returned, and then "WatchChan" is
-	// closed with non-nil "Err()".
-	// Otherwise, as long as the context has not been canceled or timed out, watch will
-	// retry on other recoverable errors forever until reconnected.
+	// In other words, a large number of watchers with "context.Background/TODO" can
```
Comment (xiang90) [inline clientv3/watch.go]: How about documenting it this way?

```
The passed in context MUST be cancelled as soon as watcher is no longer being used to release the associated resources.
```

File: clientv3/watch.go [hunk 2]
```diff
@@ -50,11 +50,13 @@ type Watcher interface {
...
+	// lead to excessive memory usage.
+	// One of the non-recoverable errors from watch API is "etcdserver: no leader"
```
Comment (xiang90) [inline clientv3/watch.go]: let us not change anything else in this PR, just add one simple sentence about the cancelling of the context.

**Comment by gyuho** [comment]
@xiang90 PTAL.

**Comment by xiang90** [comment]
lgtm. defer to @jpbetz or @spzala

**Comment by xiang90** [comment]
kindly ping @spzala

**Comment by spzala** [comment]
@xiang90 @gyuho sorry for delay, somehow I missed few notifications. Reviewing it. Thanks!

**Comment by spzala** [comment]
lgtm.

**Comment by spzala** [comment]
Since it's small PR I am taking this as opportunity to merge a first PR :-) considering lgtm from @xiang90 as well.

**Comment by spzala** [comment]
Well, no I should wait to hear from @jpbetz

**Comment by xiang90** [comment]
@spzala 

I just need one of you to give another lgtm. You can get this merged.

For larger PRs, we usually wait for the original author to click the merge button if the author is a maintainer to allow squash or further pre-merging clean up work.

**Comment by spzala** [comment]
@xiang90 thanks, that sounds great!! Merging :-)

**Pull request merged**